az-core/src/main/java/azkaban/Constants.java

```diff
@@ -195,6 +195,16 @@
 
     // dir to keep dependency plugins
     public static final String DEPENDENCY_PLUGIN_DIR = "azkaban.dependency.plugin.dir";
+
+    /*
+     * Prefix used to construct Hadoop/Spark user job link.
+     * a) RM: resource manager
+     * b) JHS: Hadoop job history server
+     * c) SHS: spark job history server
+     * */
+    public static final String AZKABAN_RM_JOB_LINK = "azkaban.rm.job.link";
```

Comment:
The '.' in the key names are used to separate namespaces, NOT to separate words. 
How about azkaban.external_resources.resource_manager? 

----
Why is it better than using the full name in the variable name e.g. RESOURCE_MANAGER_LINK ?

Reply:
What could be the full value like given this key. Any examples? @jamiesjc 
Is this a resource manager link or a job log link?
Why not call it YARN at least? It should be more straightforward.

Reply:
I'm not aware of the naming convention here. Thanks for pointing it out. 
This is more like a prefix for constructing Azkaban job logs. It's not the specific link for resource manager or job history server. So I think it's necessary to add "azkaban" and "job link" in the variable name. The reason I'm not using full name simply because it would look too long, something like AZKABAN_SPARK_JOB_HISTORY_SERVER_JOB_LINK. But I'm ok with using the full name if it's easier to understand.

Reply:
Since it's our internal configuration. I'm not able to give the details here. The value would look like: azkaban.rm.job.link=http://***rm***.linkedin.com:8088/cluster/app/
azkaban.jhs.job.link=http://***jh***.linkedin.com:19888/jobhistory/job/job_
azkaban.shs.job.link=http://***sh***.linkedin.com:18080/history/
As I said, it's a prefix for constructing the job log link. If naming it YARN, people still get confused whether the log is on resource manager or job history server.

Reply:
@HappyRay I notice we already use "azkaban.server.external" to name some external configurations, so probabaly no need for introducing the new name "external_resources"? In addition to my previous thoughts, I think it would be better to change the names to:
AZKABAN_SERVER_EXTERNAL_RM_JOB_URL = "azkaban.server.external.rm.job.url";
AZKABAN_SERVER_EXTERNAL_JHS_JOB_URL = "azkaban.server.external.jhs.job.url";
AZKABAN_SERVER_EXTERNAL_SHS_JOB_URL = "azkaban.server.external.shs.job.url";
I think using the full name would be a little too long. I've already added comments to clarify the names, so I think it should be easy to understand as well. 
Let me know if you have any concern.

Reply:
>  AZKABAN_SPARK_JOB_HISTORY_SERVER_JOB_LINK
Why is it necessary to have the Azkaban prefix?

Reply:
For Spark jobs launched from Azkaban, the job link is:
http://***sh**.grid.linkedin.com:18080/history/application_***_***/1/jobs/
It's kind of specific to Azkaban. If launched from other places, the link might vary a little. So I added the Azkaban prefix.

Reply:
But since this is the only place in our code which references that link and shouldn't cause confusion, I think it's ok to remove Azkaban prefix if it's too long.

Reply:
Discussed offline and agreed on the naming convention. See updates in the new commit.